src/Test.UnitTests.Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBaseTests.cs

```diff
@@ -1571,6 +1576,53 @@ private static Notification BuildTestNotification(Uri uri)
             };
         }
 
+        [Fact]
+        public void MultithreadedAnalyzeCommandBase_DirectoriesAreEnumerated()
+        {
+            string testDirectory = "TestDirectory";
+            string testFile = "TestFile.txt";
+            string directory = Path.GetDirectoryName(GetThisTestAssemblyFilePath());
+
+            // Create test dictionary
+            string testDirectoryPath = Path.Combine(directory, testDirectory);
+            Directory.CreateDirectory(testDirectoryPath);
+
+            // Create test file
+            string filePath = Path.Combine(testDirectoryPath, testFile);
+            if (!File.Exists(filePath))
+            {
+                File.WriteAllText(filePath, $"{Guid.NewGuid()}");
+            }
+
+            var testCases = new List<string>
+            {
+                // Absolute directory paths
+                testDirectoryPath,
+                testDirectoryPath + "/",
+                testDirectoryPath + "/*",
+
+                // Absolute file path
+                filePath,
+
+                // Relative directory paths
+                testDirectory,
+                testDirectory + "/",
+                testDirectory + "/*",
+                "./" + testDirectory,
+                "./" + testDirectory + "/",
+
+                // Relative file paths
+                Path.Combine(testDirectory, testFile),
+                "./" + Path.Combine(testDirectory, testFile),
+                $"{testDirectory}//{testFile}",
+            };
+
+            foreach (string testCase in testCases)
+            {
+                AnalyzeFile(testCase);
```

Comment:
>AnalyzeFile [](http://example.com/codeflow?start=16&length=11)

Add a comment to this like so:

// AnalyzeFile will ensure that a scan context was created for
// 'TestFile.txt`, indicating that it was successfully found at the
//  file enumeration phase (driven by 'OrderedFileSpecifier').